lib/src/symbol_maps/dart_symbol_map_debug_files_collector.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:file/file.dart';
+import 'package:sentry_dart_plugin/src/utils/flutter_debug_files.dart';
+
+import '../configuration.dart';
+
+/// Collects Flutter-relevant native debug file paths that should be paired
+/// with a Dart symbol map for symbolication.
+///
+/// Policy:
+/// - Android: include Flutter-generated `.symbols` files (e.g.,
+///   `app.android-arm.symbols`, `app.android-arm64.symbols`, `app.android-x64.symbols`).
+/// - Apple: include the Mach-O binary `App` inside
+///   `App.framework.dSYM/Contents/Resources/DWARF/App`.
+///
+/// The function returns absolute, deduplicated paths. It enumerates the
+/// configured `symbolsFolder`, `buildFilesFolder`, and other Flutter
+/// search roots discovered by `enumerateDebugSearchRoots`.
+Future<Set<String>> collectDebugFilesForDartMap({
+  required FileSystem fs,
+  required Configuration config,
+}) async {
+  final Set<String> foundPaths = <String>{};
+  final path = fs.path;
+
+  Future<void> collectAndroidSymbolsUnder(String rootPath) async {
```

Comment:
collectAndroidSymbolsUnder and containsAndroidSymbols have so much similar code, could we extract it into a higher order function?

Reply:
but yeah it shouldn't be like this